lib/datasets.go

```diff
@@ -759,11 +759,15 @@ func (r *DatasetRequests) Remove(p *RemoveParams, res *RemoveResponse) error {
 	}
 
 	// Get the revisions that will be deleted.
-	history, err := base.DatasetLog(ctx, r.node.Repo, ref, p.Revision.Gen+1, 0, false)
-	if err == nil && p.Revision.Gen >= len(history) {
+	historyDataLog, err := base.DatasetLog(ctx, r.node.Repo, ref, p.Revision.Gen+1, 0, false)
```

Comment:
I think a better (simpler) approach is change `history` to be a `[]base.DatasetLogItem`, then change `base.RemoveEntireDataset` so that it's third parameter is also a `[]base.DatasetLogItem`.